pkg/tfbridge/assets.go

```diff
@@ -156,22 +156,24 @@ func (a *AssetTranslation) TranslateAsset(asset *resource.Asset) (interface{}, e
 
 // TranslateArchive translates the given archive using the directives provided by the translation info.
 func (a *AssetTranslation) TranslateArchive(archive *resource.Archive) (interface{}, error) {
-	contract.Assert(a.IsArchive())
-
 	// TODO[pulumi/pulumi#153]: support HashField.
 
 	// Produce either a temp file or an in-memory representation, as requested.
+	format := a.Format
+	if format == resource.NotArchive {
+		format = resource.ZIPArchive
+	}
```

Comment:
Do we want to make the `ResourceInfo` author specify the desired archive kind to opt into this?  That at least ensures that this can be added incrementally without breaking changes.

Reply:
I think that accepting both an `Asset` and an `Archive` is the correct default behavior. I worry that this is subtle enough that schema authors won't know that they should set a value for `Format` when choosing an `Asset`-typed translation. I also don't think that changing the archive format is necessarily a breaking change.

Reply:
After working through the examples of how we'll use this in practice - I agree.  If we hit a case where we need more flexibility - we can extend as needed.